src/models/Automaton.java

```diff
@@ -46,6 +51,23 @@ public Automaton(String name, List<Location> locations, List<Edge> edges, List<C
             }
         }
 
+        /* The finite set of actions must be partitioned into inputs and outputs.
+         *   Here we check whether they are partitioned by checking that the intersection of the
+         *   inputs and outputs are empty as an action can only be either an input xor an output.
+         *   We don't have to check whether an action is neither an input nor an output as the set
+         *   of actions is build as the union of the inputs and outputs and for this reason
+         *   guarantees to be in the set of actions. */
+        Set<Channel> intersection = new HashSet<>(inputAct);
+        intersection.retainAll(outputAct);
+        if (!intersection.isEmpty()) {
+            throw new IllegalArgumentException("The action set of a specification must be a partition, where each action is either an input xor an output");
```

Comment:
```suggestion
            throw new IllegalArgumentException("The action set of a specification must be a partition, where each action is either an input xor an output.");
```

Or more indicative for the end user where the problem is.

```suggestion
            throw new IllegalArgumentException(String.format("The action set of specification automaton %s is not a partition.", name));
```
You could even extend this by finding a human readable way of printing all the elements of `intersection`, which could be just a single element or a lot of elements.

Reply:
I have changed it to ```throw new IllegalArgumentException(String.format("The actions %s of specification automaton %s is not a partition.", violatingActions, name));``` where ```violatingActions``` is a string joined by the names of all actions in the intersection "{a, b, c}". Is the message in the exception still correct? Maybe we should also consider encapsulating the cause as a Throwable for the IllegalArgument Exception? This could be done by creating a ```Throwable``` ```PartitionException``` and its constructor might take all actions or just the violating ones, then an instance of the ```PartitionException``` could be used as the cause of the ```IllegalArgumentException```. Overall I think the creation of a ```PartitionException``` in this case is a good idea because the ```violatingActions``` might be a very large string.

Reply:
Technically the exception text `"The actions %s of specification automaton %s is not a partition."`, because partition is a property of the complete action set. But we only want to display the actions that are causing the action set to be *not* a partition. 
Since this list of actions can vary in size (and potentiallly be very long), maybe it is more user friendly to start with the fixed length part like `"The actions set of specification automaton %s is not a partition, caused by %s."`. When we create a `PartitionException`, we can let that exception do the formatting. So it will also be easier to change if we want to.